### ISSUE TITLE
Explorer menu invisible with out of index error

### ISSUE THREAD

**Issue by abhirocks9636** [wagtail/wagtail#2580]
```
Internal Server Error: /admin/explorer-nav/
Traceback (most recent call last):
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/django/views/decorators/cache.py", line 43, in _cache_controlled
    response = viewfunc(request, *args, **kw)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/wagtail/wagtailadmin/decorators.py", line 22, in decorated_view
    return view_func(request, *args, **kwargs)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/wagtail/wagtailadmin/views/pages.py", line 27, in explorer_nav
    'nodes': get_navigation_menu_items(),
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/wagtail/wagtailcore/models.py", line 1378, in get_navigation_menu_items
    parent_page, parent_childlist = depth_list[page.depth - 1]
IndexError: list index out of range
```

**Comment by m1kola** [comment]
Hi @abhirocks9636!

Can you give more details on how to reproduce it, please?

**Comment by abhirocks9636** [comment]
It happened when i was trying to  moved some of pages from site1 root  to site2 root and deleted site1 to make site2 default after that whenever i try to open page explorer page reloads to root.
I don't understand  a bit what happened?
Please help
Thank You

**Comment by gasman** [comment]
Hi @abhirocks9636 - can you try running `./manage.py fixtree` (and paste the output here), and see if that fixes it?

**Comment by abhirocks9636** [comment]
hello @gasman ..thanks for this wonderful software..
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/wagtail/wagtailcore/management/commands/fixtree.py", line 34, in handle
    page.delete()
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/wagtail/wagtailcore/models.py", line 497, in delete
    return super(Page, self).delete(*args, **kwargs)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/treebeard/models.py", line 506, in delete
    self.__class__.objects.filter(pk=self.pk).delete()
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/treebeard/mp_tree.py", line 84, in delete
    parents[parentpath] = node.get_parent(True)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/treebeard/mp_tree.py", line 1026, in get_parent
    self.__class__).objects.get(path=parentpath)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/django/db/models/manager.py", line 122, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/var/www/vhosts/example.com/django-app/django-app-venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 387, in get
    self.model._meta.object_name
wagtail.wagtailcore.models.DoesNotExist: Page matching query does not exist.

**Comment by abhirocks9636** [comment]
still unable to fix tree structure....any help

**Comment by strindhaug** [comment]
Got the same issue on my localhost server.

Not sure how it happened, but I assume it's probably caused by some messed up back and forwards migrations while developing. (I'm using dbsqlite locally, which doesn't have transactions, and therefore messes up failed migrations badly.)

I could just have deleted my database and started over, but I didn't want to retype all my test articles, so...

I managed to force the fixtree command to complete by temporarily changing some files directly in wagtail. I post my brutal "just-ignore-the-errors" fix here in case anyone else have the same problem.
This is only doable on localhost, but then again I assume it's most likely to occur only while developing locally using dbsqlite and not in prod anyway.


My "fix" consists of changing two lines in `site-packages/wagtail/wagtailcore/models.py`.
I changed line 426 in the method `clean` from:
```python
        if not Page._slug_is_available(self.slug, self.get_parent(), self):
            raise ValidationError({'slug': _("This slug is already in use")})
```
to:
```python
        try:
            tmp = Page._slug_is_available(self.slug, self.get_parent(), self)
        except Exception as e:
            pass
        if not tmp:
            raise ValidationError({'slug': _("This slug is already in use")})
```
And this line  (what originally was line line 698 in the same file) of the method `unpublish` from:
```python
            page_unpublished.send(sender=self.specific_class, instance=self.specific)
```
to:
```python
            try:
                page_unpublished.send(sender=self.specific_class, instance=self.specific)
            except Exception as e:
                pass
```

Remember to change the file back once fix tree has completed!
